lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services
+  attr_accessor :registered_vehicles,
+                :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_details)
+    @name                 = facility_details[:name]
+    @address              = facility_details[:address]
+    @phone                = facility_details[:phone]
+    @services             = []
+    @registered_vehicles  = []
+    @collected_fees       = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      vehicle.registration_date = Date.today
+      if vehicle.antique?
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle?
+        vehicle.plate_type = :ev
+        @collected_fees += 200
+      else 
+        vehicle.plate_type = :regular
+        @collected_fees += 100
+      end
+      @registered_vehicles << vehicle
+    else
+      nil
+    end
```

Comment:
The `else` is also not strictly necessary. By default this method will return `nil` if the `if` condition is not true.